lib/add-in-repo-tests-to-host.js

```diff
@@ -5,24 +5,29 @@ const mergeTrees = require('broccoli-merge-trees');
 const Funnel = require('broccoli-funnel');
 const flattenAddons = require('./flatten-addons');
 
-const generateTreeFromPath = path =>
+const generateWatchedDirFromPath = (path) =>
   existsSync(path) ? new WatchedDir(path) : null;
 
-const getInRepoAddonTestTrees = inRepoAddons => {
+const generateTreeFromPath = (path, processTests) =>
+  existsSync(path) ? new Funnel(processTests(path)) : null;
+
+const getInRepoAddonTestTrees = (inRepoAddons, processTests) => {
   if (inRepoAddons.length === 0) {
     return [null];
   }
 
-  return inRepoAddons.map(inRepoAddon => {
+  return inRepoAddons.map((inRepoAddon) => {
     const inRepoAddonTestPath = `${inRepoAddon.root}/tests`;
 
     return existsSync(inRepoAddonTestPath)
-      ? new Funnel(inRepoAddonTestPath, { destDir: inRepoAddon.name })
+      ? new Funnel(processTests(inRepoAddonTestPath), {
+          destDir: inRepoAddon.name,
+        })
       : null;
   });
 };
 
-module.exports = (emberProject, predicate) => {
```

Comment:
nit: In some languages, underscore `_` is used to denote param will be unused. 

`processTests = _ => _` may be more clear for readability